docs/developers/sbt.md

```diff
@@ -0,0 +1,276 @@
+---
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+# Building via SBT
+
+Starting from version 0.4.0, the Celeborn project supports building and packaging using SBT. This article provides a detailed guide on how to build the Celeborn project using SBT.
+
+# System Requirements
+
+Celeborn Service (master/worker) supports Scala 2.11/Scala 2.12 and Java 8/11/17.
+
+The table following indicates the compatibility of Celeborn Spark and Flink clients with different versions of Spark and Flink for various Java and Scala versions:
+
+|                | Java 8/Scala 2.11 | Java 8/Scala 2.12 | Java 11/Scala 2.12 | Java 17/Scala 2.12 |
+|----------------|-------------------|-------------------|--------------------|--------------------|
+| Spark 2.4      | &#10004;          | &#x274C;          | &#x274C;           | &#x274C;           |
+| Spark 3.0      | &#x274C;          | &#10004;          | &#10004;           | &#x274C;           |
+| Spark 3.1      | &#x274C;          | &#10004;          | &#10004;           | &#x274C;           |
+| Spark 3.2      | &#x274C;          | &#10004;          | &#10004;           | &#x274C;           |
+| Spark 3.3      | &#x274C;          | &#10004;          | &#10004;           | &#10004;           |
+| Spark 3.4      | &#x274C;          | &#10004;          | &#10004;           | &#10004;           |
+| Flink 1.14     | &#x274C;          | &#10004;          | &#10004;           | &#x274C;           |
+| Flink 1.15     | &#x274C;          | &#10004;          | &#10004;           | &#x274C;           |
+| Flink 1.17     | &#x274C;          | &#10004;          | &#10004;           | &#x274C;           |
+
+# Useful SBT commands
+
+## Packaging the Project
+
+As an example, one can build a version of Celeborn as follows:
+
+```
+./build/sbt clean package
+```
+
+To create a Celeborn distribution like those distributed by the [Celeborn Downloads](https://celeborn.apache.org/download/) page, and that is laid out so as to be runnable, use `./build/make-distribution.sh` in the project root directory.
+
+```
+./build/make-distribution.sh --sbt
```

Comment:
have we supported it?

Reply:
no, it's a placeholder now, I intend to file a separate PR to support this feature.

Reply:
maybe `--sbt-enabled`?